dev-scripts/openshift_deploy.sh

```diff
@@ -0,0 +1,234 @@
+# ---------------------------------------
+if ([ -z "${OPENSHIFT_USERNAME+x}" ] || 
+    [ -z "${OPENSHIFT_PASSWORD+x}" ]) && 
+    [ -z "${OPENSHIFT_TOKEN+x}" ]; then echo "[CHE] **ERROR** Env var OPENSHIFT_USERNAME, OPENSHIFT_PASSWORD and OPENSHIFT_TOKEN are unset. You need to set username/password or token to continue. Aborting"; exit 1; fi
+
+if [ -z "${OPENSHIFT_ENDPOINT+x}" ]; then echo "[CHE] **ERROR**Env var OPENSHIFT_ENDPOINT is unset. You need to set it to continue. Aborting"; exit 1; fi
+if [ -z "${OPENSHIFT_NAMESPACE_URL+x}" ]; then echo "[CHE] **ERROR**Env var OPENSHIFT_NAMESPACE_URL is unset. You need to set it to continue. Aborting"; exit 1; fi
+
+# -----------------------------------
+# Logging on to the OpenShift cluster
+# -----------------------------------
+echo -n "[CHE] Logging on using OpenShift endpoint \"${OPENSHIFT_ENDPOINT}\"..."
+if [ -z ${OPENSHIFT_TOKEN+x} ]; then
+  oc login "${OPENSHIFT_ENDPOINT}" -u "${OPENSHIFT_USERNAME}" -p "${OPENSHIFT_PASSWORD}" > /dev/null
+  OPENSHIFT_TOKEN=$(oc whoami -t)
+else
+  oc login "${OPENSHIFT_ENDPOINT}" --token="${OPENSHIFT_TOKEN}"  > /dev/null
+fi
+echo "done!"
+
+# --------------------------
+# Create project (if needed)
+# --------------------------
+echo -n "[CHE] Checking if project \"${CHE_OPENSHIFT_PROJECT}\" exists..."
+if ! oc get project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null; then
+
+  if [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then echo "**ERROR** project doesn't exist on OSIO. HINT: you may have set the wrong OPENSHIFT_ID. Aborting"; exit 1; fi
+
+  echo -n "no creating it..."
+  oc new-project "${CHE_OPENSHIFT_PROJECT}"
+  ## TODO we should consider oc apply the latest http://central.maven.org/maven2/io/fabric8/online/packages/fabric8-online-che-quotas-oso/
+fi
+echo "done!"
+
+echo -n "[CHE] Switching to \"${CHE_OPENSHIFT_PROJECT}\"..."
+oc project "${CHE_OPENSHIFT_PROJECT}" &> /dev/null
+echo "done!"
+
+# -------------------------------------------------------------
+# Verify that Che ServiceAccount has admin rights at project level
+# -------------------------------------------------------------
+## TODO we should create Che SA if it doesn't exist
+## TODO we should check if che has admin rights before creating the role biding
+## TODO if we are not in minishift we should fail if che SA doesn't have admin rights
+if [ "${OPENSHIFT_FLAVOR}" == "minishift" ]; then
+  echo -n "[CHE] Setting admin role to \"che\" service account..."
+  oc login "${OPENSHIFT_ENDPOINT}" -u system:admin &> /dev/null
+  echo "apiVersion: v1
+kind: RoleBinding
+metadata:
+  name: che
+roleRef:
+  name: admin
+subjects:
+- kind: ServiceAccount
+  name: che" | oc apply -f -
+  oc login "${OPENSHIFT_ENDPOINT}" --token="${OPENSHIFT_TOKEN}"  &> /dev/null
+fi
+
+# ----------------------------------------------
+# Get latest version of fabric8-online templates
+# ----------------------------------------------
+echo -n "[CHE] Retrieving latest version of fabric8 online Che template..."
```

Comment:
it's because I had to it twice that I was thinking about my usecase (BTW I agree that it's not usual, but it would be nice to have alternative to grab templates)